src/enclave/Enclave/Limit.h

```diff
@@ -0,0 +1,24 @@
+#include <cstddef>
+#include <cstdint>
+
+#ifndef LIMIT_H
+#define LIMIT_H
+
+/** Limit helper functions */
+void count_rows_per_partition(uint8_t *input_rows, size_t input_rows_length,
+                              uint8_t **output_rows, size_t *output_rows_length);
+
+void compute_num_rows_per_partition(uint32_t limit,
+                                    uint8_t *input_rows, size_t input_rows_length,
+                                    uint8_t **output_rows, size_t *output_rows_length);
+
+void limit_return_rows(uint32_t limit,
+                       uint8_t *input_rows, size_t input_rows_length,
+                       uint8_t **output_rows, size_t *output_rows_length);
+
+void limit_return_rows(uint64_t partition_id,
+                       uint8_t *limit_rows, size_t limit_rows_length,
```

Comment:
If my understanding is correct, `limit_rows` is essentially a list of integers that represent how many rows to return at each partition. If so, perhaps we can rename these variables to `uint8_t *limits, size_t limits_length`?